### PR TITLE
Fix OTA download progress bug

### PR DESCRIPTION
**Pull request by duduita**
Adapt Meadow.Core to get the real download size from the `Content-Length` present in the HTTP response header, since currently, this information is obtained from the OTA message, which might not reflect the real package size, especially when an OTA requires updating the OS.

### CODE REVIEW

File: Source/Meadow.Core/Cloud/MeadowCloudUpdateService.cs
Comment (stevenkuhn) [inline Source/Meadow.Core/Cloud/MeadowCloudUpdateService.cs:265]: An issue I see with this is that the HTTP request has already been sent on line 243. I believe that the Range header won't have any effect here.

File: Source/Meadow.Core/Cloud/MeadowCloudUpdateService.cs
Comment (stevenkuhn) [inline Source/Meadow.Core/Cloud/MeadowCloudUpdateService.cs:247]: This will be different based on the Range header (if the Range section is placed above this). It will be the number of bytes remaining, not the total file size. I'm not sure if that matters, but just a heads up.

If you care about total file size, you'll have to have a variable that exists outside of this for-loop and set it the first time it attempts to download the file (when `retryCount` is `0`).

If you only care about bytes remaining, then this is fine. But it won't be representative of the total files size, _if_ this retry loop loops.

File: Source/Meadow.Core/Cloud/MeadowCloudUpdateService.cs
Comment (duduita) [reply Source/Meadow.Core/Cloud/MeadowCloudUpdateService.cs:265]: Yeah, thanks! I need to adjust it given that now we are making the request on line 243 indeed.

**Comment by alexischr** [comment]
In my testing, this fixed https://github.com/WildernessLabs/Meadow_Issues/issues/775 ! Will wait for a review from @stevenkuhn , but LGTM

**Review by alexischr — APPROVED** [review]

**Comment by stevenkuhn** [comment]
> In my testing, this fixed [WildernessLabs/Meadow_Issues#775](https://github.com/WildernessLabs/Meadow_Issues/issues/775) ! Will wait for a review from @stevenkuhn , but LGTM

Code looks great to me and awesome to have configuration that it fixes the issue! Nice work! 🎉

**Review by stevenkuhn — APPROVED** [review]

**Pull request merged**
